cnd/src/db/created_swap.rs

```diff
@@ -142,14 +137,131 @@ impl Load<CreatedSwap<herc20::CreatedSwap, halight::CreatedSwap>> for Sqlite {
             alpha,
             beta,
             peer,
-            address_hint,
             role,
         };
 
         Ok(Some(created))
     }
 }
 
+#[async_trait]
+impl Save<CreatedSwap<hbit::CreatedSwap, herc20::CreatedSwap>> for Sqlite {
+    async fn save(
+        &self,
+        created: CreatedSwap<hbit::CreatedSwap, herc20::CreatedSwap>,
+    ) -> anyhow::Result<()> {
+        let local_swap_id = created.swap_id;
+        let role = created.role;
+
+        let swap = InsertableSwap::new(created.swap_id, created.peer.clone(), created.role);
+        self.save_swap(&swap).await?;
+
+        // Save the hbit details.
+        let redeem_identity = match role {
+            Role::Alice => None,
+            Role::Bob => Some(created.alpha.identity),
+        };
+        let refund_identity = match role {
+            Role::Alice => Some(created.alpha.identity),
+            Role::Bob => None,
+        };
+
+        let hbit = InsertableHbit {
+            swap_id: 0, // FK, set during save.
+            amount: Text(created.alpha.amount.into()),
+            network: Text(created.alpha.network),
+            hash_function: Text(HashFunction::Sha256),
+            redeem_identity: redeem_identity.map(|pk| Text(pk.into())),
+            refund_identity: refund_identity.map(|pk| Text(pk.into())),
+            ledger: Text(Ledger::Alpha),
+        };
+        self.save_hbit(local_swap_id, &hbit).await?;
```

Comment:
🤔 Could be extracted into a function to reduce duplication with function below. Or at least scoped. That way the comment on line 159 can go away too.

Reply:
Same for saving herc20 details.

Reply:
Nice, will do.

Reply:
Had a play with this; it actually got me to somewhere I wanted to be without previously know how :)

We can add another type that groups with all the insertable/queryable types i.e.,
- `Foo`, `InsertableFoo`, and `FooData`

Open to a better name than `FooData`, the obvious is to use `Foo` and use `QueryableFoo` for the queryable type but this is out because of how the `belongs_to` macro works, well at least for the `Swap` type, and I'd like to be uniform.

Anyways, leaving this for a follow up PR because it touches all the db code.

Reply:
What you have touched on here @lucas is actually a really big deal, fixing this makes the whole db code better. Easier to read, write, extend, and maintain. I was looking for this, thanks.

https://github.com/comit-network/comit-rs/issues/2635